src/graph_API.jl

```diff
@@ -0,0 +1,90 @@
+type edge
+  # param_node: the "in" node (listed first in connectparameter)
+  # var_node: the "out" node (listed second in connectparameter)
+  name::Symbol
+  param_node::Symbol #name of component
+  var_node::Symbol #name of component
+end
+
+type ModelGraph
+  edges
+  nodes
+  size
+end
```

Comment:
Yes, just the length of nodes. Wasn't sure if it would be faster to keep it stored as a variable, but I imagine that asking for the length of an array is not a very costly operation, so I can delete this field if you want